src/prefect/infrastructure/docker.py

```diff
@@ -285,7 +286,18 @@ def _build_container_settings(
         )
 
     def _create_and_start_container(self) -> "Container":
-        docker_client = self._get_client()
+        if self.image_registry:
+            # If an image registry block was supplied, load an authenticated Docker
+            # client from the block. Otherwise, use an unauthenticated client to 
+            # pull images from public registries.
+            docker_client = self.image_registry.login()
+            # When using a private registry, the image name must include the registry address,
+            # even when using Docker Hub.
+            registry_address = re.sub("^https?://", "", self.image_registry.registry_url)
+            self.image = (self.image if self.image.startswith(registry_address) 
+                     else f"{registry_address}/{self.image}")
```

Comment:
I added that mostly for Docker Hub, because users accustomed to working with non-private images from Docker Hub are likely to just include the image name. When I tried that with a private Docker Hub registry, I wasn't able to pull images. Prefixing it with 'registry.hub.docker.com' solved the problem.

Maybe this is better solved with documentation than code, though.

Reply:
Good point, I'll let @desertaxle chime in to judge what's the best approach here

Reply:
I think we can address the Docker Hub nuances in documentation. I think we should attempt to make this block as generally useable as possible. We can offer more specific implementations if necessary.